include/material_system/material.hpp

```diff
@@ -59,6 +64,12 @@ namespace GraphicsEngine {
 
         [[nodiscard]] virtual Material* clone() const noexcept;
 
+        static constexpr uint32_t serial_version = 0x1;
+
+	    void serialize(ByteWriter& writer);
+
+	    static std::shared_ptr<Material> deserialize(ByteReader& reader);
```

Comment:
why not materialserializer class method?

Reply:
a nah mu4it jopy esli 1) serializer nujen polniy dostup v predstavlenie classa
esli sdelat friend to ne 04 + pridetsya dlya kajdoy template instantiation pisat
2) objecty poh *kak* ego serializiruyut. emy ne poh tolko *4to* serializiruyut

tipa mogy soglasitsya na templated serialize + deserialize methody vnutri cla𐌔𐌔a (i to pridetsya delat tresh default constructor ibo ty j ne ho4ew constructor iz readera)
no otdelniy interface dlya etogo eto perebor